bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeConverter.java

```diff
@@ -275,32 +275,37 @@ public static Set<Type> getConvertibleTypes(Object inputValue, Type targetType,
                     if ((inputValueType == memType) || isIntegerSubtypeAndConvertible(inputValue, memType)) {
                         return Set.of(memType);
                     }
-                    convertibleTypes.addAll(getConvertibleTypes(inputValue, memType, varName,
-                            isFromJson, unresolvedValues, errors, allowAmbiguity));
+                    convertibleTypes.addAll(getConvertibleTypes(inputValue, memType, varName, isFromJson,
+                            unresolvedValues, errors, allowAmbiguity, numericTypeSet, targetType));
                 }
-                if (!allowAmbiguity && (convertibleTypes.size() > 1) && !convertibleTypes.contains(inputValueType) &&
-                        !hasIntegerSubTypes(convertibleTypes)) {
-                    errors.subList(initialErrorCount, errors.size()).clear();
-                    addErrorMessage(0, errors, "value '" + getShortSourceValue(inputValue)
-                            + "' cannot be converted to '" + targetType + "': ambiguous target type");
-                    return new LinkedHashSet<>();
+                if (convertibleTypes.size() > 1) {
```

Comment:
Shall we check whether we can write like https://github.com/HindujaB/ballerina-lang/pull/1? Shall we also add tests with tuple type rest types, and 2D arrays to check whether recursive calls are working?